kaggle_environments/agent.py

```diff
@@ -112,20 +112,32 @@ def act(self, observation, timeout=10):
         if self.agent is None:
             self.agent = build_agent(self.raw, self.environment)
             # Add in the initialization timeout since this is the first time this agent is called
-            timeout += self.configuration.agentTimeout
+            timeout += self.environment.configuration.agentTimeout
 
         args = [
            structify(observation),
-           structify(self.configuration)
+           structify(self.environment.configuration)
         ][:self.agent.__code__.co_argcount]
 
-        try:
-            action = self.agent(*args)
-        except Exception as e:
-            action = e
+        with StringIO() as out_buffer, StringIO() as err_buffer, redirect_stdout(out_buffer), redirect_stderr(err_buffer):
+            try:
+                action = self.agent(*args)
+            except Exception as e:
+                action = e
+            # Allow up to 1k log characters per step which is ~1MB per 600 step episode
+            max_log_length = 1024
+            out = out_buffer.getvalue()[0:max_log_length]
+            err = err_buffer.getvalue()[0:max_log_length]
+
+        duration = time() - start
+        log = {
+            "duration": round(duration, 6),
```

Comment:
floating point numbers have more precision than time() so we trim off the extra digits here. Windows users get even less precision (1/16 ms vs 1 microsecond in Linux / OS X).

Reply:
I usually use time.perf_counter() as it usually uses a reference point that is closer in time.